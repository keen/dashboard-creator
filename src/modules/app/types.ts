export type ViewMode = 'editor' | 'viewer' | 'management';

export type ReducerState = {
  view: ViewMode;
  activeDashboardId: string | null;
  cachedDashboardsNumber: number;
  user: {
    editPrivileges: boolean;
  };
  imagePicker: {
    isVisible: false;
  };
  visualizationEditor: {
    isVisible: false;
  };
  queryPicker: {
    isVisible: boolean;
  };
};
