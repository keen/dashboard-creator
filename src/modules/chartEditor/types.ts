import { Query } from '@keen.io/query';
import {
  PickerWidgets,
  ChartSettings,
  WidgetSettings,
} from '@keen.io/widget-picker';
import { TextSettings } from '@keen.io/widgets';

export enum EditorSection {
  QUERY = 'query',
  SETTINGS = 'settings',
}

export type EditorWidgetSettings = WidgetSettings & {
  title: TextSettings;
  subtitle: TextSettings;
};

export type ReducerState = {
  editorSection: EditorSection;
  isOpen: boolean;
  isEditMode: boolean;
  isSavedQuery: boolean;
  isDirtyQuery: boolean;
  isQueryPerforming: boolean;
  hasQueryChanged: boolean;
  changeQueryConfirmation: boolean;
  querySettings: Partial<Query>;
  initialQuerySettings: Partial<Query>;
  analysisResult: Record<string, any> | null;
  queryError: string | null;
  visualization: {
    type: Exclude<PickerWidgets, 'json'>;
    chartSettings: ChartSettings;
    widgetSettings: EditorWidgetSettings;
  };
};

export type VisualisationSettingsPayload = {
  type: PickerWidgets;
  chartSettings: ChartSettings;
  widgetSettings: Record<string, any>;
};
