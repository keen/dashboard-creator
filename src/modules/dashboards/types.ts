import { Theme } from '@keen.io/charts';
import { Widget } from '../widgets';

export type DashboardMetaData = {
  id: string;
  title: null | string;
  widgets: number;
  queries: number;
  tags: string[];
  lastModificationDate: number;
  isPublic: boolean;
  publicAccessKey: null | string;
};

export type DashboardSettings = {
  colorPalette: string;
  page: {
    gridGap: number;
    background: string;
    chartTitlesFont: string;
    visualizationsFont: string;
  };
  tiles: {
    background: string;
    borderColor: string;
    borderRadius: number;
    borderWidth: number;
    padding: number;
    hasShadow: boolean;
  };
};

export type DashboardModel = {
  version: string;
  widgets: Widget[];
  settings?: DashboardSettings;
  theme?: Partial<Theme>;
};

export type Dashboard = {
  version: string;
  widgets: string[];
};

export enum DashboardError {
  NOT_EXIST = 'NOT_EXIST',
  ACCESS_NOT_PUBLIC = 'ACCESS_NOT_PUBLIC',
  VIEW_PUBLIC_DASHBOARD = 'VIEW_PUBLIC_DASHBOARD',
}

export type DashboardItem = {
  initialized: boolean;
  isSaving: boolean;
  settings: Dashboard;
  error: null | DashboardError;
};

export type DashboardListOrder = 'recent' | 'oldest' | 'az' | 'za';

export type ReducerState = {
  deleteConfirmation: {
    isVisible: boolean;
    dashboardId?: string;
  };
  metadata: {
    isInitiallyLoaded: boolean;
    error: boolean;
    data: DashboardMetaData[];
    isSavingMetadata: boolean;
    isRegeneratingAccessKey: boolean;
  };
  dashboardSettingsModal: {
    isVisible: boolean;
    dashboardId: string | null;
  };
  dashboardShareModal: {
    isVisible: boolean;
    dashboardId: string | null;
  };
  tagsPool: string[];
  tagsFilters: {
    showOnlyPublicDashboards: boolean;
    tags: string[];
  };
  items: Record<string, DashboardItem>;
  dashboardListOrder: DashboardListOrder;
  cachedDashboardIds: string[];
};
