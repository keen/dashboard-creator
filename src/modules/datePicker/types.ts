export type ReducerState = {
  isEditorOpen: boolean;
  widgetConnections: DatePickerConnection[];
};

export type DatePickerConnection = {
  widgetId: string;
  isConnected: boolean;
  title: string | null;
  positionIndex: number;
};
