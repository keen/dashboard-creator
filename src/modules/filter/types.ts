export type ReducerState = {
  isEditorOpen: boolean;
  widgetConnections: FilterConnection[];
  eventStream: string;
  targetProperty: string;
};

export type FilterConnection = {
  widgetId: string;
  isConnected: boolean;
  title: string | null;
  positionIndex: number;
};
