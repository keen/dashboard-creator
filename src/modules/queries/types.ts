import { Query } from '@keen.io/parser';

export type SavedQueryAPIResponse = {
  query: Query;
  query_name: string;
};

export type SavedQuery = {
  id: string;
  settings: Query;
};
