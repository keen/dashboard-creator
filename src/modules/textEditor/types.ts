import { RawDraftContentState } from 'draft-js';

export type ReducerState = {
  isOpen: boolean;
  textAlignment: TextAlignment;
  content: RawDraftContentState;
};

export type TextAlignment = 'left' | 'center' | 'right';
