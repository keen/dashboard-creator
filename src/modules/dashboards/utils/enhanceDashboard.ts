import deepMerge from 'deepmerge';
import { Theme } from '@keen.io/charts';

import createDashboardSettings from './createDashboardSettings';

import { DashboardModel } from '../types';

/**
 * Enhances dashboard model with properties. Used for providing
 * backward compatibility,
 *
 * @param dashboard - Dashboard model
 * @param baseTheme - Base theme connected with partial provided in configuration
 * @return dashboard model with extended properties
 *
 */
const enhanceDashboard = (dashboard: DashboardModel, baseTheme: Theme) => {
  return {
    ...dashboard,
    ...('theme' in dashboard
      ? {}
      : {
          theme: baseTheme,
        }),
    ...('settings' in dashboard
      ? {
          settings: deepMerge(createDashboardSettings(), dashboard.settings, {
            arrayMerge: (_target, source) => source,
          }),
        }
      : {
          settings: createDashboardSettings(),
        }),
  };
};

export default enhanceDashboard;
