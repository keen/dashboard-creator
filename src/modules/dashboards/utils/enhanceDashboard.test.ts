import enhanceDashboard from './enhanceDashboard';
import { DashboardModel } from '../types';
import { Theme } from '@keen.io/charts';

test('Enhances dashboard model with provided properties', () => {
  const dashboard = {
    settings: {
      title: {
        typography: {
          fontStyle: 'normal',
          fontColor: 'red',
        },
      },
      subtitle: {
        typography: {
          fontStyle: 'italic',
          fontColor: 'green',
        },
      },
    },
  } as DashboardModel;

  const baseTheme = {
    colors: ['yellow', 'orange'],
  } as Theme;

  expect(enhanceDashboard(dashboard, baseTheme)).toMatchSnapshot();
});
