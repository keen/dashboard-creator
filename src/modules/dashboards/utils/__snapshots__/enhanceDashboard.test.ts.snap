// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enhances dashboard model with provided properties 1`] = `
Object {
  "settings": Object {
    "colorPalette": "default",
    "legend": Object {
      "typography": Object {
        "fontColor": "#1D2729",
        "fontFamily": "Gangster Grotesk, Arial, Helvetica, sans-serif",
        "fontSize": 10,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    },
    "page": Object {
      "background": "#f1f5f8",
      "chartTitlesFont": "Gangster Grotesk",
      "gridGap": 20,
      "visualizationsFont": "Lato",
    },
    "subtitle": Object {
      "typography": Object {
        "fontColor": "green",
        "fontFamily": "Gangster Grotesk, Arial, Helvetica, sans-serif",
        "fontSize": 14,
        "fontStyle": "italic",
        "fontWeight": "normal",
      },
    },
    "tiles": Object {
      "background": "#ffffff",
      "borderColor": "none",
      "borderRadius": 0,
      "borderWidth": 0,
      "hasShadow": true,
      "padding": 20,
    },
    "title": Object {
      "typography": Object {
        "fontColor": "red",
        "fontFamily": "Gangster Grotesk, Arial, Helvetica, sans-serif",
        "fontSize": 20,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    },
  },
  "theme": Object {
    "colors": Array [
      "yellow",
      "orange",
    ],
  },
}
`;
